test/verify/check-pkgupdates

```diff
@@ -0,0 +1,445 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+from testlib import *
+
+@skipImage("Image uses OSTree", "continuous-atomic", "fedora-atomic", "rhel-atomic")
+@skipImage("PackageKit crashes, https://launchpad.net/bugs/1689820", "ubuntu-1604")
+class TestUpdates(MachineCase):
+    def setUp(self):
+        MachineCase.setUp(self)
+
+        self.isApt = "debian" in self.machine.image or "ubuntu" in self.machine.image
+
+        # disable all existing repositories to avoid hitting the network
+        if self.isApt:
+            self.machine.execute("rm -f /etc/apt/sources.list.d/*; echo > /etc/apt/sources.list; apt-get update")
+        else:
+            self.machine.execute("rm -f /etc/yum.repos.d/* /var/cache/yum/*")
+
+        # have PackageKit start from a clean slate
+        self.machine.execute("systemctl stop packagekit; rm -rf /var/cache/PackageKit")
+
+        if self.machine.image in ["ubuntu-1604", "debian-stable"]:
+            # old PackageKit+NM on Debian/Ubuntu misdetect online status with ifupdown; work around
+            # https://launchpad.net/bugs/1694438
+            # TODO: use a naughty override instead?
```

Comment:
Can do, but this means that we wouldn't test OS updates on debian-stable *at all*. It wouldn't affect debian-stable users who use ifupdown (i. e. servers), or use NetworkManager without a static ifupdown config (i. e. desktops). This just affects the case of ifupdown *plus* NM that we use on our test images. So I thought it would be more useful to run the checks with a config that works than to skip them entirely due to our special setup.